vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -724,6 +729,39 @@ void AdsImpl::OnPageLoaded(
       << previous_tab_url_;
 }
 
+void AdsImpl::ExtractPurchaseIntentSignal(const std::string& url) {
+  if (!ShouldClassifyPagesIfTargeted()) {
+    return;
+  }
+
+  // TODO(Terry Mancey): might fail TLD+1
+  if (!TestSearchState(url) &&
+      helper::Uri::MatchesDomainOrHost(url, previous_tab_url_)) {
+    return;
+  }
+
+  PurchaseIntentSignalInfo purchase_intent_signal =
+      purchase_intent_classifier_->ExtractIntentSignal(url);
+
+  if (purchase_intent_signal.segments.empty() &&
+      purchase_intent_signal.timestamp_in_seconds == 0) {
+    return;
+  }
+
+  GeneratePurchaseIntentSignalHistoryEntry(purchase_intent_signal);
+  BLOG(INFO) << "Intent signal extracted for " << url;
```

Comment:
nit: Can we please change to "Purchase intent signal extracted for " just to make logs more clear for support?